azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -398,10 +403,71 @@ public void createContainer(final int executionId) throws ExecutorManagerExcepti
    */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    deletePod(executionId);
-    if (isServiceRequired()) {
-      deleteService(executionId);
+    try { // if pod deletion is not successful, the service deletion can still be handled
+      deletePod(executionId);
+    } finally {
+      if (isServiceRequired()) {
+        deleteService(executionId);
+      }
+    }
+  }
+
+  /**
+   * This method is used to delete invalid containers in batch
+   * @param containerValidity pod validity in milliseconds
+   * @throws ExecutorManagerException
+   */
+  @Override
+  public void deleteAgedContainers(Duration containerValidity) throws ExecutorManagerException {
+    logger.info(String.format("Cleaning up containers older than %d days",
+        containerValidity.toDays()));
+    final Set<Integer> containersToDelete = getStaleContainers(containerValidity);
+    for (int execId : containersToDelete) {
+      if (execId > 0) {
+        logger.info(String.format("Deleting stale pod %s and service %s", getPodName(execId),
+            getServiceName(execId)));
+        deleteContainer(execId);
```

Comment:
I think we should handle exceptions thrown by `deleteContainer` method here. For example, lets say there are 3 containers to delete but first deletion attempt failed, because an exception was thrown we're not going to attempt deletion of the remaining 2. Moreover, the cleanup routine will run periodically and if the list of containers to delete is sorted the same every time only we'll probably never pass the first attempt.

Reply:
Added a try catch block to handle exception thrown by deleteContainer.